src/com/github/alex_moon/porcupi/contexts/PokeContext.java

```diff
@@ -0,0 +1,49 @@
+package com.github.alex_moon.porcupi.contexts;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import com.github.alex_moon.porcupi.handlers.PokeHandler;
+import com.github.alex_moon.porcupi.handlers.Pokeable;
+import com.github.alex_moon.porcupi.messages.PokeContextMessage;
+
+
+public class PokeContext extends Context {
+    private String key;
+    private Pokeable pokeable;
+
+    public PokeContext(long tid, PokeHandler handler, Pokeable pokeable, String key) {
+        super(tid, handler);
+        this.context = "poke";
+        this.key = key;
+        this.pokeable = pokeable;
+    }
+    
+    public String getKey() {
+        return key;
+    }
+    
+    public void handleMessage(String message) {
+        super.handleMessage(message);
```

Comment:
Not sure I'm a fan of this - it would be nice to do a method called by the superclass instead... just an idea